### PR TITLE
chore: refactoring to makes all things work together [DHIS2-13232]

### PR DESCRIPTION
**Pull request by gnespolino**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/linelisting/trackedentityinstance/TeiAnalyticsQueryService.java [hunk 1]
```diff
@@ -65,18 +64,23 @@
...
-        final SqlQueryResult result = (SqlQueryResult) queryExecutor.execute( query );
-        final List<GridHeader> headers = from( ((SqlQuery) query).getColumns() );
+        final SqlQuery query = (SqlQuery) teiJdbcQuery.from( teiLineListingParams );
+        final SqlQueryResult result = queryExecutor.execute( query );
+        final List<GridHeader> headers = from( query.getColumns() );
 
         return gridAdaptor.createGrid( headers, result.result() );
     }
+
+    public void validateRequest( TeiLineListingRequest request )
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/linelisting/trackedentityinstance/TeiAnalyticsQueryService.java:82]: In our last discussion, I think we decided to move it into a separate component.

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/linelist/trackedentity/TrackedEntityLineListingController.java [hunk 2]
```diff
@@ -59,15 +63,15 @@
     @GetMapping( "query/{trackedEntityType}" )
     Grid getGrid(
         @PathVariable String trackedEntityType,
-        TrackedEntityLineListingRequest request,
+        TeiLineListingRequest request,
```
Comment (maikelarabori) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/linelist/trackedentity/TrackedEntityLineListingController.java:66]: I think we decided to pass in two different request objects, so we could remove the class extension.

**Review by maikelarabori — COMMENTED** [review]
Looks good @gnespolino, I just raised two points we had discussed recently. Please take a look.

Also, I'm not sure that we should use "line list/listing" in the backend. This endpoint can be used by any front-end application/client. I don't think it should be tight to a specific application.

Thanks,
Maikel

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/linelisting/trackedentityinstance/TeiAnalyticsQueryService.java (hunk 1 above)
Comment (gnespolino) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/linelisting/trackedentityinstance/TeiAnalyticsQueryService.java:82]: Ok, I'll extract it into a dedicated tei validation service

**Pull request merged**